decidim-system/app/forms/decidim/system/oauth_application_form.rb

```diff
@@ -1,18 +1,18 @@
 # frozen_string_literal: true
 
 module Decidim
-  module Admin
+  module System
     # The form that validates the data to construct a valid OAuthApplication.
     class OAuthApplicationForm < Decidim::Form
       mimic :oauth_application
 
       attribute :name, String
-      attribute :organization_name, String
+      attribute :decidim_organization_id, Integer
       attribute :organization_url, String
       attribute :organization_logo
       attribute :redirect_uri, String
 
-      validates :name, :redirect_uri, :current_user, :current_organization, :organization_name, :organization_url, :organization_logo, presence: true
+      validates :name, :redirect_uri, :decidim_organization_id, :organization_url, :organization_logo, presence: true
```

Comment:
I'd suggest adding a method to load the organization object, and use that from the command instead of the id. 

I prefer using objects rather than ids since the id is a persistence detail, and also to avoid possible errors or data inconsistency.